api/query/query.go

```diff
@@ -93,16 +107,25 @@ func (qh queryHandler) loadSummaries(testRuns shared.TestRuns) ([]summary, error
 			defer wg.Done()
 
 			var data []byte
-			s := make(summary)
+			s := summary{
+				Old: nil,
+				New: nil,
+			}
 			data, loadErr := qh.loadSummary(testRun)
 			if err == nil && loadErr != nil {
 				err = fmt.Errorf("Failed to load test run %v: %s", testRun.ID, loadErr.Error())
 				return
 			}
-			marshalErr := json.Unmarshal(data, &s)
+			// Try to unmarshal the json using the new aggregation structure.
+			marshalErr := json.Unmarshal(data, &s.New)
```

Comment:
A thought for myself: summary files are big and this might increase the overheads of loading runs. Either add a flag somewhere to indict the old or new, or do a page load test

Reply:
Yeah, I was reading up about using this approach to check if the summary is old or new, but I'm not certain it's the most efficient. I guess It might be worth doing some timing to figure out what kind of time this adds.